features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/route/engine/condition/engine/InsertClauseShardingConditionEngine.java

```diff
@@ -100,7 +100,7 @@ private void appendMissingShardingConditions(final InsertStatementContext sqlSta
             return;
         }
         for (String each : allColumnNames) {
-            if (!columnNames.contains(each) && !shardingRule.isGenerateKeyColumn(each, tableName) && shardingRule.findShardingColumn(each, tableName).isPresent()) {
+            if (!columnNames.contains(each.toLowerCase()) && !shardingRule.isGenerateKeyColumn(each, tableName) && shardingRule.findShardingColumn(each, tableName).isPresent()) {
```

Comment:
Why convert each to lowerCase?

Reply:
https://user-images.githubusercontent.com/1734396/255540089-b905d899-c0e7-4053-abc3-f9f7df27054a.png
this method make all field to lowercase, but SS not process the field of table, so there is a problem when table contain case-sensitive fields, even though the field is same one  ,such as 'accountNo'

Reply:
In my opinion, adapting column identifier to `lowercase` is not good choice. It looks just avoiding this issue.
We need to fix `getColumnNamesForInsertColumns` returns it's own identifier, not `lowercase`.
When just fixing with this way, it will be appeared on other case later.

Reply:
Yes, all modifications are to fix this problem, not avoid it. I asked this question in the previous issue, and the answer at that time was "ShardingSphere adopts a case-insensitive strategy, which is matched in lowe Rcase"

Reply:
Aha, Thank you!